Define gates by matrix or permutation (#8)
rigetti/quil: spec/Quil.md

    -i*sin(%theta/2), cos(%theta/2)
```

If the matrix can be represented as a permutation, then the gate can

> be defined with the compact notation

Reviewer: ```suggestion
be defined with the compact notation:
```

```

> DEFGATE NAME AS PERMUTATION:

Reviewer: ```suggestion
DEFGATE Name AS PERMUTATION:
```

>     P_1, P_2, ..., P_N

Reviewer: ```suggestion
    P_0, P_1, ..., P_(N-1)
```

```

> with each `P_i` being a strictly positive integer, and `N×N` being the

Reviewer: ```suggestion
with each `P_i` being a non-negativbe integer, and `N×N` being the
```

dimension of the intended matrix representation. See the `CCNOT` example

> below.

Reviewer: ```suggestion
below. The values of `P_0` to `P_(N-1)` indicate how the bits of the bitstring are permuted: The zeroth bit goes to `P_0`, the first bit goes to `P_1`, and so on.
```
